### ISSUE TITLE
CPqD switch seems to have changed `dpctl` so that `UserSwitch.connected()` fails

### ISSUE THREAD

**Issue by lantz** [mininet/mininet#745]
### Expected/Desired Behavior:

Ideally it would work, but now CPqD `ofsoftswitch` is now incompatible with `UserSwitch`'s `dpctl`.

### Actual Behavior:

mininet> py s1.connected()
*** s1 : ('dpctl status tcp:127.0.0.1:6654',)
dpctl: Error connecting to switch status. (Address family not supported by protocol)

### Detailed Steps to Reproduce the Behavior:
`sudo mn --controller ryu,simple_switch_13.py --switch user -v debug --test pingpair`

**Comment by cheriimoya** [comment]
At the moment, installing with `mininet/util/install.sh -3fy` fails.

But after testing using [this pr](https://github.com/cheriimoya/mininet/tree/test/test-pr) (which combines https://github.com/mininet/mininet/pull/1104 and https://github.com/mininet/mininet/pull/1105) for the installation with `mininet/util/install.sh && mininet/util/install.sh -3fy`, `sudo mn --controller ryu,simple_switch_13.py --switch user -v debug --test pingpair`.

I have no clue what it should do, but it prints a lot of information over and over
```
...
*** s1 : ('dpctl status tcp:127.0.0.1:6654',)
config.management0=ptcp:6654
config.probe-interval=15
config.max-backoff=15
switch.now=1643799572
switch.uptime=8
switch.pid=42500
local.name=unix:/tmp/s1
local.state=ACTIVE
local.backoff=1
local.is-connected=true
local.sent-msgs=1
local.received-msgs=11
local.attempted-connections=2
local.successful-connections=1
local.last-connection=7
local.time-connected=7
local.state-elapsed=7
remote.name=tcp:127.0.0.1:6653
remote.state=BACKOFF
remote.backoff=8
remote.is-connected=false
remote.sent-msgs=0
remote.received-msgs=0
remote.attempted-connections=4
remote.successful-connections=0
remote.last-connection=8
remote.time-connected=0
remote.state-elapsed=1
in-band.local-mac=00:00:00:00:00:01
...
```

Running `sudo mn --controller ryu,simple_switch_13.py --switch user` and then `py s1.connected()` doesn't throw an error though

**Comment by davidcawork** [comment]
Indeed, as @lantz comments, Eder's tool, BOFUSS repo (https://github.com/CPqD/ofsoftswitch13), `dpctl` no longer supports the `status` command. 

The tool originally has a CLI and Eder rewrote it, so there is no compatibility between the two... This is very important to differentiate, as the original and eder's do NOT work the same. I was unaware of this. And the problem is that resulting binary is called the same, and from the BOFUSS repository, the [manpages of the old tool are installed](https://github.com/CPqD/ofsoftswitch13/blob/master/utilities/dpctl.8.in). So we have two different versions of `dpctl` tool:

- Original `dpctl`: https://github.com/mininet/openflow/blob/master/utilities/dpctl.c
- Eder version `dpctl`: https://github.com/CPqD/ofsoftswitch13/blob/master/utilities/dpctl.c


---

With this in mind, we can see substantial differences by running the `dpctl -h` command, common to both of them. One of the most important differences is the order in which the switch identifier is indicated, either by a tcp port, unix socket... Below I indicate what I indicate for each version:

```bash
# Original
sudo dpctl CMD [SW]
```

```bash
# Eder version
sudo dpctl [SW] CDM
```

It is also worth mentioning that the available commands are completely different between the two versions. For example, one of the most commonly used commands is `dump-flows` which is **not** found in the new version of Eder's tool. 

![a](https://raw.githubusercontent.com/davidcawork/TFM/main/doc/img/1.png)

Here is a screenshot of the commands available for the new version of the tool. For example, the analogous command to the old `dpctl dump-flows [SW]` is `dpctl [SW] stats-flow`. 


---

Taking this into account. If at the time of installation:

-    If `-0` is used, the repository (https://github.com/mininet/openflow) will be installed, using the original `dpctl` tool. 
-    If `-3` is used, the repository (https://github.com/CPqD/ofsoftswitch13) will be installed, using Eder's version `dpctl` tool.


The order of the indicated parameters will be important in this case, since it will be the last of them that will dictate to `-f` param which version of the `dpctl` tool prevails. 

---

In summary for this issue. For the original `dpctl` tool this command will work. For the Eder tool (`dpctl version`), it will not work. Evidence of this behaviour is attached below. I've used a scenario with Vagrant, which raises an Ubuntu server 22.04 machine, and installs Mininet.

*    Scenario used: https://github.com/davidcawork/TFM/tree/main/src/scenarios/mininet

The installation parameters used are the following: 

```bash
./mininet/util/install.sh -3fmnyv
```

In this case, the Eder version of the tool is installed, and the command proposed in the issue does not have to work:

```bash
sudo mn --controller ryu,simple_switch_13.py --switch user -v debug
```

![ss](https://raw.githubusercontent.com/davidcawork/TFM/main/doc/img/2.png)


If we install the original tool: 

```bash
# Clone the repo
git clone https://github.com/mininet/openflow.git

# Go ahead
cd openflow

# config automake
./boot.sh
./configure

# Build
make

# Lets install the original dpctl tool :)
sudo make install
```

And if we replicate the same test, we can see that we have the expected behaviour.

![](https://raw.githubusercontent.com/davidcawork/TFM/main/doc/img/3.png)


Hope this workaround helps someone :smile: !!
